core/computers.go

```diff
@@ -107,3 +109,20 @@ func GetPercentageOfValue(value *big.Int, percentage float64) *big.Int {
 func SafeMul(a uint64, b uint64) *big.Int {
 	return big.NewInt(0).Mul(big.NewInt(0).SetUint64(a), big.NewInt(0).SetUint64(b))
 }
+
+// SafeSubUint64 performs subtraction on uint64 and returns an error if it overflows
+func SafeSubUint64(a, b uint64) (uint64, error) {
```

Comment:
unit tests?

Reply:
We can let these unit tests for someone else

Reply:
done